src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -892,9 +892,9 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
 
                     return typeRef;
                 }
-                else if (namedTypeSymbol.IsNativeInt)
+                else if (namedTypeSymbol.NativeIntegerUnderlyingType is NamedTypeSymbol underlyingType)
                 {
-                    return namedTypeSymbol.AsNativeInt(asNativeInt: false);
+                    return underlyingType;
```

Comment:
>return underlyingType; [](start = 20, length = 22)

It is probably better to adjust namedTypeSymbol and fall through. It look like there is an expectation that definitions do not return early.